ide/extexecution.base/test/unit/src/org/netbeans/api/extexecution/base/ExplicitProcessParametersTest.java

```diff
@@ -0,0 +1,322 @@
+        assertTrue("Args must be replaced by default", extra.isArgReplacement());
+        assertNull(extra.getPriorityArguments());
+
+        
+        assertContains(p.getArguments(), "avalanche");
+        assertNotContains(p.getArguments(), "File1");
+    }
+    
+    public void testSingleDefaultLaunchAugmentation() throws Exception {
+        ExplicitProcessParameters extra = ExplicitProcessParameters.builder().
+                arg("avalanche").
+                priorityArg("-Dfoo=bar").
+                build();
+        
+        assertFalse("No prio override requested", extra.isPriorityArgReplacement());
+        assertTrue("Args must be replaced by default", extra.isArgReplacement());
+        assertNotNull(extra.getPriorityArguments());
+        assertNotNull(extra.getArguments());
+
+        ExplicitProcessParameters p = ExplicitProcessParameters.builder().
+                priorityArgs(existingVMArgs).
+                args(existingAppArgs).
+                combine(extra).
+                build();
+        
+        assertContains(p.getPriorityArguments(), "-Xmx100m", "-Dfoo=bar");
+        assertContains(p.getArguments(), "avalanche");
+        assertNotContains(p.getArguments(), "File1");
+    }
+    
+    /**
+     * Checks that VM parmeters can be replaced.
+     * @throws Exception 
+     */
+    public void testReplacePriorityArgs() throws Exception {
+        ExplicitProcessParameters extra = ExplicitProcessParameters.builder().
+                priorityArg("-Dfoo=bar").
+                appendPriorityArgs(false).
+                build();
+        
+        ExplicitProcessParameters extra2 = ExplicitProcessParameters.builder().
+                priorityArg("-Dsun=shines").
+                build();
+
+        assertTrue("Must replace priority args", extra.isPriorityArgReplacement());
+        assertFalse("No arguments were specified", extra.isArgReplacement());
+        assertNull(extra.getArguments());
+
+        
+        ExplicitProcessParameters p = ExplicitProcessParameters.builder().
+                priorityArgs(existingVMArgs).
+                args(existingAppArgs).
+                combine(extra).
+                build();
+        
+        assertContains(p.getPriorityArguments(), "-Dfoo=bar");
+        assertNotContains(p.getPriorityArguments(), "-Xmx100m");
+        
+        ExplicitProcessParameters p2 = ExplicitProcessParameters.builder().
+                priorityArgs(existingVMArgs).
+                args(existingAppArgs).
+                combine(
+                    ExplicitProcessParameters.buildExplicitParameters(Arrays.asList(extra, extra2))
+                ).
+                build();
+        
+        assertContains(p2.getPriorityArguments(), "-Dfoo=bar", "-Dsun=shines");
+        assertNotContains(p2.getPriorityArguments(), "-Xmx100m");
+    }
+    
+    public void testAppendNormalArgs() throws Exception {
+        ExplicitProcessParameters extra = ExplicitProcessParameters.builder().
```

Comment:
Some of these tests could be referenced with the `{@codesnippet}` tag in the documentation.

Reply:
Documented in 39417f0d8b